iothub/service/tests/IotHubConnectionPropertiesTests.cs

```diff
@@ -18,6 +19,7 @@ public class IotHubConnectionPropertiesTests
         [DataRow("acme2.azure-devices.net", "acme2")]
         [DataRow("3acme.azure-devices.net", "3acme")]
         [DataRow("4-acme.azure-devices.net", "4-acme")]
+        [DataRow("5acme", "5acme")]
```

Comment:
Q - what scenario is this datarow testing?

Reply:
This tests a conditional expression:

`string iotHubName = index >= 0
                ? hostName.Substring(0, index)
                : hostName;`

It looks like we try and find a HostNameSeperator (".") and if we can't, then we just return the whole thing. The last case there tests the latter case.